docs/shrc.md

```diff
@@ -66,43 +140,138 @@ The [-Xshareclasses](xshareclasses.md) option is highly configurable, allowing y
+
+The following main functions are available from the `SharedClassHelper` API:
+
+- `findSharedClass`: Used to check whether a class is already in the cache before looking for the class on the filesystem.
+- `storeSharedClass`: Used to store a class in the cache.
+- `setSharingFilter`: A filter that can be used to decide which classes are found and stored in the cache. This filter can be applied to a particular package. To apply a filter to all non-bootstrap class loaders that share classes, specify the `-Dcom.ibm.oti.shared.SharedClassGlobalFilterClass` system property on the command line.
+
+The following APIs are available for obtaining information about a shared classes cache:
 
-Several utilities are provided for managing active caches, which are invoked by specifying `-Xshareclasses` suboptions. These utilities control the following types of operations:
+- `com.ibm.oti.shared.SharedClassStatistics`: Obtains information about cache size, including free space, soft maximum limit, and the limits enforced for AOT and JIT data.
+- `com.ibm.oti.shared.SharedClassUtilities`: Obtains detailed information about a shared classes cache, including its size, name, type, and status.
+- `com.ibm.oti.shared.SharedClassCacheInfo`: Stores information about a shared classes cache and provides API methods to retrieve the information and remove caches. For example, you can use the `IterateSharedCaches` and `DestroySharedCache` [JVMTI extensions](interface_jvmti.md#iteratesharedcaches).
 
-- Adjust the minimum and maximum amount of cache space reserved for AOT or JIT data.
-- Adjust the soft maximum size of the cache.
-- Create snapshots of a cache.
-- Create a cache from a snapshot.
-- Remove caches and cache snapshots.
-- List all the compatible and incompatible caches and snapshots.
-- For problem determination, invalidate and revalidate AOT methods that cause a failure in an application.
-- For problem determination, provide information to analyze the contents of a shared classes cache.
+Each class loader that wants to share data must get a `SharedDataHelper` object from a `SharedDataHelperFactory`.
+A `SharedDataHelperFactory` provides an interface that can be used to create `SharedDataHelpers`, which are used for storing Java byte array data. A `SharedDataHelper` also has a one to one relationship with a class loader, although a class loader can exist without a `SharedDataHelper`.
 
-For more information, see [-Xshareclasses](xshareclasses.md).
 
 ## See also
 
 - [AOT compiler](aot.md)
-- [Class Sharing](https://www.ibm.com/developerworks/java/library/j-ibmjava4/index.html)
+- [Class sharing article](https://www.ibm.com/developerworks/java/library/j-ibmjava4/index.html)
```

Comment:
This link doesn't work any more. I think https://developer.ibm.com/languages/java/tutorials/j-class-sharing-openj9/ is the latest.